### ISSUE TITLE
Please archive the project instead of deleting it.

### ISSUE THREAD

**Issue by NathanSMB** [fafhrd91/actix-web-postmortem#5]
> At the moment I am planing to make repos private and then delete them (will remove benchmarks as well), unless others suggest better ideas.

I think archiving would be a better idea here.

1) Some companies and developers may rely on those repos. Deleting them will burn bridges. I understand your done with open source but deleting is the nuclear option here.

2) Archiving makes the project readonly and lets others know it won't be updated. This seems to be the main goal anyway.

Here is the github documentation for archiving. I hope you choose this path and I wish you luck in the future.

https://help.github.com/en/github/creating-cloning-and-archiving-repositories/archiving-a-github-repository

Edit: Another recommendation would be to copy the last version of actix-web to this repo and then archive this repo with the rest. Continue new work on your personal github but keep an archived version here.

**Comment by artursapek** [comment]
Yeah, this is a bigger "fuck you" to the community/ecosystem than was warranted

**Comment by joaoBeno** [comment]
Boy, what a situation... Whatever the author decide to do, I'm fine with it... It is his code after all... I just got started on rust, and this a big cautionary tale about Rust... If businesses depend on the code, they should contact the author and license it (Even a free license is a license), so they can use and distribute it...

**Comment by NathanSMB** [comment]
> and this a big cautionary tale about Rust
~ @joaoBeno
 
Why? What about this situation would have been prevented if this project was in another language?

You are not alone in your view which is another reason I believe archiving this project is the best solution. It reflect negatively on the whole ecosystem if a project of this size shuts down.

And yes the community was toxic. This happens in all communities of a significant size. We should not ignore it though and I hope the Rust community managers are brain storming how to mitigate the kind of behavior we saw the past 24 hours.

**Comment by RyanBluth** [comment]
An alternative solution might be to move the project to Gitlab and lock down who can create issues and open merge requests. That way @fafhrd91 can control who has access and avoid being harassed in issue threads. It would be a shame to see such a great project stop progressing due to a toxic subset of the Rust community

**Comment by easyaspi314** [comment]
Can agree. While the community was being rude, deleting it was not the right response. This will break every project that depends on it.

Currently, it is still up on his personal account, https://github.com/fafhrd91/actix-web, and I mirrored it at https://github.com/easyaspi314/actix-web

While I don't intend to maintain it, at least it should be safe there.

**Comment by NathanSMB** [comment]
@RyanBluth He can continue the project wherever he wants. I still think there should still be an archived version here so there aren't dead links in other projects.

**Comment by RyanBluth** [comment]
@NathanSMB Agreed. I think Archiving here would be great. Just suggesting Gitlab as an option in case he wants to continue development while making the code accessible in a more controlled environment.

**Comment by kfogel** [comment]
> At the moment I am planing to make repos private and then delete them (will remove benchmarks as well), unless others suggest better ideas.

The first commenter, @NathanSMB, makes an excellent suggestion.  Why not just leave the code in place here, announce that you're no longer maintaining it, and let others clone copies as they wish so that at least others can continue?  You can ask them not to call it "actix-web" if you want, to avoid reputational entanglement, but there's no need to take the code itself (and its version-control history) away.

I hope someone had a recent clone of the repository and will just post it so work can continue.

**Comment by easyaspi314** [comment]
> I hope someone had a recent clone of the repository and will just post it so work can continue

As I literally posted 42 minutes ago, I have a copy at https://github.com/easyaspi314/actix-web, and the original repo seems to be available at https://github.com/fafhrd91/actix-web

I don't have intentions of being a maintainer though (I know very little Rust), I am more concerned about preservation at the minute.

**Comment by artemonster** [comment]
> Yeah, this is a bigger "fuck you" to the community/ecosystem than was warranted

yeah, fuck you! warranted.

**Comment by cup** [comment]
https://github.com/fafhrd91/actix-web/issues/1284

**Comment by kosinix** [comment]
Oh come on give the guy a break. We're still asking him to do something for you and the community. Let those who need the code do the hard work this time.

**Comment by miguelmota** [comment]
Open source != free labor

Open source authors are free to do as they wish with their own projects.

**Comment by joaoBeno** [comment]
> 
> 
> > and this a big cautionary tale about Rust
> > ~ @joaoBeno
> 
> Why? What about this situation would have been prevented if this project was in another language?

The drama around the developer design choices... He decided to something a certain way, most communities would just fork it and do it "the right way™", and leave the author alone... 

I said that this is a cautionary tale in the sense that people may be less inclined to show something that they think is great, on the fear that, after sometime, someone decides that all was done wrong and the Dev must eat shit for it...

> 
> You are not alone in your view which is another reason I believe archiving this project is the best solution. It reflect negatively on the whole ecosystem if a project of this size shuts down.

Sure it is, and people mirroring it should read the license to be sure that they have the right to do so when the author decides to pull his work from the net... This was a very bad situation, and much has to be learnt from it...

> 
> And yes the community was toxic. This happens in all communities of a significant size. We should not ignore it though and I hope the Rust community managers are brain storming how to mitigate the kind of behavior we saw the past 24 hours.

Sure, but toxic people are hard to change... I think that a hard lesson today may leave a long lasting memory, so when the next Dev does things "the wrong way", people hold more and talk more before going for the pitchforks...
